src/main/java/org/terasology/web/serverAdminManagement/AdminPermissionManager.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.serverAdminManagement;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import javafx.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.paths.PathManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.permission.PermissionManager;
+import org.terasology.network.Client;
+import org.terasology.network.ClientComponent;
+import org.terasology.network.NetworkSystem;
+import org.terasology.registry.In;
+import org.terasology.web.resources.DefaultComponentSystem;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Manages the serverAdminPermissions.json file in the server directory.
+ */
+public final class AdminPermissionManager implements DefaultComponentSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(AdminPermissionManager.class);
+    private static final Gson GSON = new Gson();
+    private static AdminPermissionManager instance;
+
+    @In
+    private NetworkSystem networkSystem;
+    @In
+    private PermissionManager permissionManager;
+
+    private final Path adminPermissionsFilePath;
+    private final Type typeOfServerAdminPermissions = new TypeToken<Set<Pair<String, Map<PermissionType, Boolean>>>>() {
+    }.getType();
+    private Set<Pair<String, Map<PermissionType, Boolean>>> serverAdminPermissions;
```

Comment:
Is there a particular reason for using `Map<PermissionType, Boolean>` instead of simply a `Set<PermissionType>`?

Reply:
I need to store the permission true/false values somewhere in `serverAdminPermissions`, so I thought it would be easier to understand and code by using a map.